### ISSUE TITLE
Sporadic Failure while starting Segmentstore due to RocksDB initialization error

### ISSUE THREAD

**Issue by shrids** [pravega/pravega#3363]
**Problem description**
Observing that the Segment store 
```
2019-02-09 05:24:07,101 995  [main] ERROR i.p.s.server.host.ServiceStarter - Could not start the Service, Aborting.
java.lang.UnsatisfiedLinkError: /tmp/librocksdbjni7887950603317182160.so: Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by /tmp/librocksdbjni7887950603317182160.so)
        at java.lang.ClassLoader$NativeLibrary.load(Native Method)
        at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
        at java.lang.Runtime.load0(Runtime.java:809)
        at java.lang.System.load(System.java:1086)
        at org.rocksdb.NativeLibraryLoader.loadLibraryFromJar(NativeLibraryLoader.java:78)
        at org.rocksdb.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:56)
        at org.rocksdb.RocksDB.loadLibrary(RocksDB.java:64)
        at org.rocksdb.RocksDB.<clinit>(RocksDB.java:35)
        at io.pravega.segmentstore.storage.impl.rocksdb.RocksDBCacheFactory.<init>(RocksDBCacheFactory.java:51)
        at io.pravega.segmentstore.server.host.ServiceStarter.lambda$attachRocksDB$1(ServiceStarter.java:165)
        at io.pravega.segmentstore.server.store.ServiceBuilder.getSingleton(ServiceBuilder.java:323)
```

**Problem location**
Pravega docker build (probably)

**Suggestions for an improvement**
Investigate the issue.

**Comment by adrianmo** [comment]
According to https://github.com/facebook/rocksdb/issues/3848, this problem is manifested when running RocksDB on Alpine with openjdk. It seems that Alpine uses musl-libc but it's not supported by RocksDB. This is the issue to add support for musl-libc to RocksDB: https://github.com/facebook/rocksdb/pull/3143

**Comment by shrids** [comment]
Observed that this error comes when the docker image is based on Alpine 3.9 
ref: https://github.com/docker-library/openjdk/commits/master/8/jre/alpine/Dockerfile
```
We can observed that the failed image  depends on image `6c40cc604d8e` which points to alpine 3.9
docker pull alpine:3.9
3.9: Pulling from library/alpine
6c40cc604d8e: Already exists 
Digest: sha256:b3dbf31b77fd99d9c08f780ce6f5282aba076d70a513a8be859d8d3a4d0c92b8
Status: Downloaded newer image for alpine:3.9

docker pull build-docker-repo:8116/nautilus/pravega:issue-3358-bugfix-okhttp-0.5.0-2116.86bf89f
issue-3358-bugfix-okhttp-0.5.0-2116.86bf89f: Pulling from nautilus/pravega
6c40cc604d8e: Pull complete 
e78b80385239: Pull complete 
f41fe1b6eee3: Pull complete 
e5cb75e950df: Pull complete 
0540af836fff: Pull complete 
ac56eb2276c6: Pull complete 
6bfea0e804aa: Pull complete 
Digest: sha256:c7a340601191799d323185b38e3a40cb5a5f147519b639a0f62bcd0377ed7fb4

The working docker image depends on `cd784148e348` image
docker pull build-docker-repo:8116/nautilus/pravega:0.5.0-2116.4c1912d
0.5.0-2116.4c1912d: Pulling from nautilus/pravega
cd784148e348: Already exists 
35920a071f91: Already exists 
1a5149a464dd: Already exists 
d8e0c842dddd: Pull complete 
3314b0e80434: Pull complete 
586fa45e5a95: Pull complete 
2c80bd142c94: Pull complete 
```

**Comment by adrianmo** [comment]
I've been able to fix this issue by installing the [`gcompat`](https://pkgs.alpinelinux.org/package/edge/community/x86/gcompat) Alpine package during the Docker build process. This package allows non-musl compatible libraries like RocksDB to run in musl environments.

According to [this blog post](https://alpinelinux.org/posts/Alpine-Linux-has-switched-to-musl-libc.html) on the AlpineLinux site, Alpine switched from glibc to musl, so that explains why RocksDB suddenly stopped working after the base image was updated.

I'll submit a PR with the fix.
